### PR TITLE
chore: per-file compiler output snapshots

### PR DESCRIPTION
**Pull request by MarkMcCulloh**
vitest recently added a feature to separate snapshots into individual files. With this, the new snapshots look beautiful (proper syntax highlighting and no more escapes!)

The main downside is that now there are a lot more files.

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: tools/hangar/__snapshots__/test_corpus/api.w/compile/tf-aws/main.tf.json [hunk 1]
```diff
@@ -121,8 +118,8 @@ exports[`wing compile -t tf-aws > main.tf.json 1`] = `
             "uniqueId": "root_cloudApi_cloudApiOnRequeste46e5cb7_IamRolePolicy_0281983F",
           },
         },
-        "policy": "{\\"Version\\":\\"2012-10-17\\",\\"Statement\\":[{\\"Action\\":[\\"dynamodb:UpdateItem\\"],\\"Resource\\":[\\"\${aws_dynamodb_table.root_cloudCounter_E0AC1263.arn}\\"],\\"Effect\\":\\"Allow\\"},{\\"Action\\":[\\"dynamodb:GetItem\\"],\\"Resource\\":[\\"\${aws_dynamodb_table.root_cloudCounter_E0AC1263.arn}\\"],\\"Effect\\":\\"Allow\\"}]}",
```
Comment (Chriscbr) [inline tools/hangar/__snapshots__/test_corpus/api.w/compile/tf-aws/main.tf.json:124]: Less escaping here is nice

**Review by Chriscbr — COMMENTED** [review]
Getting better syntax highlighting on each of the files seems like an improvement to me. I'm down to try this.

**Review by hasanaburayyan — COMMENTED** [review]
Ill take a lot more files for better readability!

**Review by Chriscbr — APPROVED** [review]

**Comment by eladb** [comment]
I thought this is cool initially, but now we have over 500 snapshot files in our repo and GitHub PRs are really huge ([example](https://github.com/winglang/wing/pull/2076)). Are we certain this is an improvement in our experience? I am okay to give it a shot for a while but I am not sure that's a better situation.

**Pull request merged**